src/main/java/com/palettee/user/service/UserService.java

```diff
@@ -0,0 +1,242 @@
+package com.palettee.user.service;
+
+import com.palettee.archive.domain.*;
+import com.palettee.archive.repository.*;
+import com.palettee.gathering.repository.*;
+import com.palettee.portfolio.domain.*;
+import com.palettee.portfolio.repository.*;
+import com.palettee.user.controller.dto.request.*;
+import com.palettee.user.controller.dto.response.*;
+import com.palettee.user.domain.*;
+import com.palettee.user.exception.*;
+import com.palettee.user.repository.*;
+import java.util.*;
+import java.util.function.*;
+import java.util.stream.*;
+import lombok.*;
+import lombok.extern.slf4j.*;
+import org.springframework.stereotype.*;
+import org.springframework.transaction.annotation.*;
+
+@Slf4j
+@Service
+@RequiredArgsConstructor
+@Transactional(readOnly = true)
+public class UserService {
+
+    private final UserRepository userRepo;
+    private final RelatedLinkRepository relatedLinkRepo;
+    private final PortFolioRepository portFolioRepo;
+    private final StoredProfileImageUrlRepository storedProfileImageUrlRepo;
+    private final ArchiveRepository archiveRepo;
+    private final GatheringRepository gatheringRepo;
+
+    /**
+     * 유저 프로필 정보 반환하는 메서드
+     * <p>
+     * 만약 자신의 프로필을 조회하면 {@link UserDetailResponse#role} 은 {@code non-null} 값으로 응답
+     *
+     * @param userId       프로필 조회할 유저 id
+     * @param loggedInUser 지금 로그인한 유저
+     * @throws UserNotFoundException id 에 해당하는 유저가 없을 때
+     */
+    public UserDetailResponse getUserDetails(Long userId, Optional<User> loggedInUser)
+            throws UserNotFoundException {
+        // 조회하려는 유저 info
+        User userOnTarget = this.getUserById(userId);
+
+        // 포폴 링크
+        String portfolioLink = this.getPortfolioLink(userOnTarget);
```

Comment:
여기서 N+1 문제가 발생하지 않나요??? Fetch join해서 프록시 초기화 안시켜주면 문제가 발생할 거 같습니다!

Reply:
전체적으로 유저를 가져오는데 fetch Lazy 방식 사용하고 있어서 fetch join으로 가져오지 않으면 N+1문제가 발생할 거 같은데.. 아닌가요??? 한번 확인 부탁드립니다!

Reply:
N+1 발생할 것 같습니다!
아마도 데이터가 적긴해서 성능상 엄청난 이슈를 없을거 같긴한데 발생하긴 할 것 같습니다!

Reply:
그럼 혹시 아래처럼 [`findAllByUserId`](https://github.com/prgrms-web-devcourse-final-project/WEB1_1_ZeroOne_BE/blob/03885a7cb71ce3d8f198720502e805d425f71f9b/src/main/java/com/palettee/portfolio/repository/PortFolioRepository.java#L10C4-L11C50) 를 사용한다면 `N + 1` 문제는 없을까요?

아니면 `findAllByUserId` 에 `fetch join` 을 넣어주면 바로 해결 될까요...?

```suggestion
        String portfolioLink = portFolioRepo.findAllByUserId(userOnTarget.getId());
```

제가 `JPA` 를 잘 몰라서 헷갈리네요 ㅠㅠ...

Reply:
네네! 그렇게 해도 N+1은 발생하지 않을 것 같습니다!

Reply:
https://github.com/prgrms-web-devcourse-final-project/WEB1_1_ZeroOne_BE/pull/52/commits/098886f7cbe227a0cc3912bd62be11741cf187f8 로 변경하였습니다!

`JPA` 도 데브코스 끝나면 깊숙히 공부해야 겠습니다 ㅠㅠ..

Reply:
고생 많으셨습니다 !!!